consensus/core/src/test_dag_builder.rs

```diff
@@ -107,6 +107,46 @@ impl DagBuilder {
         }
     }
 
+    pub(crate) fn blocks(&self, rounds: RangeInclusive<Round>) -> Vec<VerifiedBlock> {
+        assert!(
+            !self.blocks.is_empty(),
+            "No blocks have been created, please make sure that you have called build method"
+        );
+        self.blocks
+            .iter()
+            .filter_map(|(block_ref, block)| rounds.contains(&block_ref.round).then_some(block))
+            .cloned()
+            .collect::<Vec<VerifiedBlock>>()
+    }
+
+    pub(crate) fn leader_blocks(
```

Comment:
Wonder if you could replicate the logic in `test_dag::get_all_uncommitted_leader_blocks` here. Then we can remove that method. It will also allow us to get the leaders for pipeline/multileader case as well

Reply:
Good point, let me look on that